designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Dtos.scala

```diff
@@ -0,0 +1,709 @@
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object ChangedProcessingMode extends ScenarioActivityType {
+      override def displayableName: String        = "Processing mode change"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object IncomingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Incoming migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    case object OutgoingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Outgoing migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedSingleExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedScheduledExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object AutomaticUpdate extends ScenarioActivityType {
+      override def displayableName: String        = "Automatic update"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    override def values: immutable.IndexedSeq[ScenarioActivityType] = findValues
+
+  }
+
+  implicit def scenarioActivityTypeSchema: Schema[ScenarioActivityType] =
+    enumSchema[ScenarioActivityType](
+      ScenarioActivityType.values.toList,
+      _.entryName,
+    )
+
+  implicit def scenarioActivityTypeCodec: circe.Codec[ScenarioActivityType] = circe.Codec.from(
+    Decoder.decodeString.emap(str =>
+      ScenarioActivityType.withNameEither(str).left.map(_ => s"Invalid scenario action type [$str]")
+    ),
+    Encoder.encodeString.contramap(_.entryName),
+  )
+
+  implicit def scenarioActivityTypeTextCodec: Codec[String, ScenarioActivityType, CodecFormat.TextPlain] =
+    Codec.string.map(
+      Mapping.fromDecode[String, ScenarioActivityType] {
+        ScenarioActivityType.withNameOption(_) match {
+          case Some(value) => DecodeResult.Value(value)
+          case None        => DecodeResult.InvalidValue(Nil)
+        }
+      }(_.entryName)
+    )
+
+  def enumSchema[T](
```

Comment:
It looks like a generic tapir helper, maybe we could place it in some more generic place?

Reply:
moved to TapirCodecs utils

Reply:
ok, but it's not removed from this place yet